### PR TITLE
Quarkus-1405 / Mutiny 1.x improvements

### PR DESCRIPTION
**Pull request by pjgg**
Mutiny 1.x improvements, basically talks about `Mutiny context`. This PR is covering this `mutiny context` scenario

### CODE REVIEW

File: QUARKUS-1405.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Quarkus-1405 - Mutiny 1.x improvements
```
Comment (rsvoboda) [inline QUARKUS-1405.md:1]:
```suggestion
# QUARKUS-1405 - Mutiny 1.x improvements
```

File: QUARKUS-1405.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Scope of the testing
+
+Mutiny contexts shall be primarily used to share transient data used for networked I/O processing such as correlation identifiers, tokens, etc. They should not be used as general-purpose data structures that are frequently updated and that hold large amounts of data.
+
+Test development will focus on cover the notion of mutiny `context` and how to pass this metadata values between several components.
+
+### Impact on testsuites and testing automation:
+
+Reusing some existing scenarios as `Rest-client-reactive` where are involved two or more components, could be a good approach. In the end, the most important is to be sure that the context is propagated between several components.
```
Comment (rsvoboda) [inline QUARKUS-1405.md:15]:
```suggestion
Reusing some existing scenarios as `http/rest-client-reactive` where are involved two or more components, could be a good approach. In the end, the most important is to be sure that the context is propagated between several components.  
```

Comment (rsvoboda) [inline QUARKUS-1405.md:15]: context is propagated between several components ... what type of data will you propagate?

https://github.com/smallrye/smallrye-mutiny/pull/776/files#diff-0a4ca8cde3da2b020a94dbe2b82e05eec9684e2492c69ac4c356529edea82216R36 uses String

Other types could be checked too, maybe custom objects?

File: QUARKUS-1405.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+Mutiny contexts shall be primarily used to share transient data used for networked I/O processing such as correlation identifiers, tokens, etc. They should not be used as general-purpose data structures that are frequently updated and that hold large amounts of data.
+
+Test development will focus on cover the notion of mutiny `context` and how to pass this metadata values between several components.
+
+### Impact on testsuites and testing automation:
+
+Reusing some existing scenarios as `Rest-client-reactive` where are involved two or more components, could be a good approach. In the end, the most important is to be sure that the context is propagated between several components.  
+
+### Advanced topics for test development
```
Comment (rsvoboda) [inline QUARKUS-1405.md:17]: Assuming these are topics for the future development. Correct?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-1405.md (hunk 2 above)
Comment (pjgg) [reply QUARKUS-1405.md:15]: in theory, based in the context API, other objects are supported: 

```
public Context put(String key, Object value)
```

File: QUARKUS-1405.md (hunk 3 above)
Comment (pjgg) [reply QUARKUS-1405.md:19]: Yes, well, If we have enough time some of these topics could be covered in this cycle.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**